File: doc-examples/example-java/src/test/java/example/UserRepositorySpec.java
package example;

import io.micronaut.context.annotation.Property;
import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import javax.inject.Inject;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

@MicronautTest
@Property(name = "jpa.default.properties.hibernate.hbm2ddl.auto", value = "create-drop")
class UserRepositorySpec {
    @Inject UserRepository userRepository;

    @Test
    void testSoftDelete() {
        final User joe = new User();
        joe.setName("Joe");
        final User fred = new User();
        fred.setName("Fred");
        final User bob = new User();
        bob.setName("Bob");
        userRepository.saveAll(Arrays.asList(
                fred,
                bob,
                joe
        ));

        userRepository.deleteById(joe.getId());

        assertEquals(2, userRepository.count());
        assertTrue(userRepository.existsById(fred.getId()));
        assertFalse(userRepository.existsById(joe.getId()));

        final List<User> disabled = userRepository.findDisabled();
        assertEquals(1, disabled.size());
        assertEquals("Joe", disabled.iterator().next().getName());
    }
}

File: doc-examples/jdbc-example-kotlin/src/test/kotlin/example/UserRepositorySpec.kt
package example

import io.micronaut.test.annotation.MicronautTest
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.*

@MicronautTest
class UserRepositorySpec (val userRepository: UserRepository) {

    @Test
    fun testSoftDelete() {
        val joe = User(0, "Joe", true)
        val fred = User(0, "Fred", true)
        val bob = User(0, "Bob", true)
        userRepository.saveAll(listOf(fred, bob, joe))

        userRepository.deleteById(joe.id!!)

        assertEquals(2, userRepository.count())
        assertTrue(userRepository.existsById(fred.id!!))
        assertFalse(userRepository.existsById(joe.id!!))

        val disabled = userRepository.findDisabled()
        assertEquals(1, disabled.size)
        assertEquals("Joe", disabled.first().name)
    }
}
